src/core/types.h

```diff
@@ -788,7 +788,8 @@ int binarySearch(T needle, RandomAccessIterator start, RandomAccessIterator end,
     int l = 0;
     int r = end - start - 1;
     while (l <= r) {
-        int mid = l + (r - l) / 2;
```

Comment:
Thanks @wangjb13  i fully understand what you'r saying.
so i think i need to revert my commit about 'src/core/types.h'
do you know the way to revert it?